addons/stock/views/product_views.xml

```diff
@@ -97,11 +99,88 @@
         </record>
 
         <!-- view common to both template and product -->
-        <record id="view_template_property_form" model="ir.ui.view">
+        <record id="product_template_form_view" model="ir.ui.view">
             <field name="name">product.template.stock.property.form.inherit</field>
             <field name="model">product.template</field>
+            <field name="priority">20</field>
             <field name="inherit_id" ref="product.product_template_form_view"/>
             <field name="arch" type="xml">
+                <header position="inside">
+                    <button string="Update Quantity" type="object"
+                        groups="stock.group_stock_manager"
+                        name="action_update_quantity_on_hand"
+                        attrs="{'invisible': [('type', '!=', 'product')]}"/>
+                    <button string="Replenish" type="action"
+                        name="%(action_product_replenish)d"
+                        context="{'default_product_id': id}"
+                        groups="stock.group_stock_user"
+                        attrs="{'invisible': [('type', '!=', 'product')]}"/>
+                </header>
+                <div name="button_box" position="inside">
+                    <button class="oe_stat_button"
+                           name="action_open_quants"
+                           icon="fa-cubes"
+                           type="object" attrs="{'invisible':[('type', '!=', 'product')]}">
+                           <div class="o_field_widget o_stat_info">
+                                <span class="o_stat_value">
+                                    <field name="qty_available" widget="statinfo" nolabel="1" class="mr4"/>
+                                    <field name="uom_name"/>
+                                </span>
+                                <span class="o_stat_text">On Hand</span>
+                           </div>
+                    </button>
+                    <button type="object"
+                        name="action_product_forecast_report"
+                        attrs="{'invisible':[('type', '!=', 'product')]}"
+                        class="oe_stat_button" icon="fa-cubes">
+                        <div class="o_field_widget o_stat_info">
+                            <span class="o_stat_value">
+                                <field name="virtual_available" widget="statinfo" nolabel="1" class="mr4"/>
+                                <field name="uom_name"/>
+                            </span>
+                            <span class="o_stat_text">Forecasted</span>
+                        </div>
+                    </button>
+                   <button string="Traceability"
+                        type="object"
+                        name= "action_view_stock_move_lines"
+                        attrs="{'invisible':[('type', 'not in', ['product', 'consu'])]}"
```

Comment:
Just a feedback (Across all)
instead of checking `not in` `['product', 'consu']`  cant we just check `= 'service'` only
(Or Am I missing something here  ?)

Reply:
Well, I'm just reordering/cleaning stuff here, I want to limit the changes as much as possible.
I don't have another type in mind (from my limited knowledge) but I think it is better to target the only types who should see this button instead of targeting the types who shouldn't see it.  It ensures another type added who knows where doesn't have access to functionalities it shouldn't have access to.

Reply:
> It ensures another type added who knows where doesn't have access to functionalities it shouldn't have access to.

So for the same reason adding` = 'service' `will support this case. (and Support desired behaviour when someone is going to add new type in future, But by adding `not in` one has top change domain everywhere to support the use case)

Ps: I know it is same at every other place, so Let's keep it as it is.

Reply:
Well, the stock is build to manage moves of products either consumable or storable, so we can consider that if any other product type is added, we don't want to show/do any stock flow for it.  Therefore, by adding `not in product/consu`, we ensure it is shown only to those two types we support and not any other one which could be added any further (because only those two are supported by the stock anyway).  If we set it to `= service`, any custom type of product would see stock actions, even if they are not supported for the given type.

Reply:
Well, we can also have a use case which is completely on another side of this. 
But let's keep as it is for now as It's not gonna happen (At least not in near future)